.github/workflows/build.yaml

```diff
@@ -50,7 +50,7 @@ jobs:
 
       - name: Checkout required submodules
         if: steps.smcache.outputs.cache-hit != 'true'
-        run: git submodule update --init -j $(nproc) --depth 1 $(echo ${submodule_paths} | sed '$d' | tr '\n' ' ')
+        run: git submodule update --init -j $(nproc) $(echo ${submodule_paths} | sed '$d' | tr '\n' ' ')
```

Comment:
As mentioned in the PR conversation I don't think that this is the correct solution since the shallow clone/checkout was deliberately added recently for a specific purpose:

- https://github.com/riscv-collab/riscv-gnu-toolchain/pull/1608
  - https://github.com/riscv-collab/riscv-gnu-toolchain/commit/efc7cc9c828050d99125b0b6054a4ab80901adc8

Ditto for the removal of `shallow = true` in `.gitmodules` below.